lua/gruvbox_crisp/groups.lua

```
@@ -0,0 +1,314 @@
+  end
+  if not allowed_contrast[contrast] then
+    contrast = "highest"
+  end
+
+  -- Base background derived from validated contrast, with transparent override
+  local base_bg = (contrast == "soft") and p.bg1 or p.bg0
+  if transparent then base_bg = p.none end
+
+  -- Prepare surface layers for downstream use
+  -- In soft contrast, lift secondary surfaces so they remain distinct
+  local surface0, surface1, surface2, surface3
```

Luacheck (1.2.0)

[warning] variable 'surface0' is never accessed

(W231)